docs/interfaces/action-service.txt

```diff
@@ -1,27 +1,19 @@
 # Action Service Interface
 
-// TODO: More Exceptions?
-Exception ActionNotFound(name: string);
+Exception ActionException(message: string);
 
 
 /**
  * Executes multiple actions in the conext of the user given by the user_id.
+ * All actions are successful if no exception is thrown, else no action was
+ * processed.
  *
- * @throws ActionNotFound
+ *
+ * @throws ActionException
  */
-execute(actions: Action[], user_id: number): ExecuteResult;
+handle_request (payload: Action[], user_id: number): void
 
 interface Action {
-    name: string;
-    data: object;
-}
-
-interface ActionError {
-    error: string;
-    argumens: string[];
+    action: string;
+    data: object[];
```

Comment:
Why must the data be an array and cannot be an object? Maybe you should provide examples, of how you designed some specific actions.

Reply:
Data should be an array because you should be able to do every action as mass/bulk action.
For every action there is a json schema in the code the defines the expected payload, e. g. https://github.com/OpenSlides/openslides-backend/blob/master/openslides_backend/actions/topic/create.py#L16

Reply:
Ah, I had something like this in mind. Maybe you document this here with a comment and e.g. a `Type ActionData=object` and using `ActionData` as a clear name.

Reply:
Changed.